### DISCUSSION TITLE
Message lost when use share subscription by java paho client API

### DISCUSSION THREAD

**Discussion by SomberOfShadow** [emqx/emqx#6821] [Q&A]
hello, all
I use java paho client API to connect EMQ X cluster（opensource version）, and subscribe topic, then get the messages arrived .
However, I encounter a question which confused me .

I use thread pool to create multi client(create one client in each thread ) to connect EMQ X cluster, but I found sometimes not all messages can arrived . I did not find the reason now. 

The Shared subscription  is a new feature in MQTT 5.0 and I see  paho-java does not support MQTT5.0 well from below link:

https://www.emqx.io/docs/zh/v4.3/development/java.html#paho-java-mqtt-5-0-%E6%94%AF%E6%8C%81

I am not sure whether it is the root cause.

So , I will appreciate it if anyone can give me any help.

Best Regards

**Comment by SomberOfShadow** [comment]
The java paho dependency is as below:
        <dependency>
```
            <groupId>org.eclipse.paho</groupId>
            <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
            <version>1.2.5</version>
        </dependency>
```

**Comment by k32** [comment]
When shared subscription is used, the messages are forwarded to one of the clients. If the client disconnects, the messages sent to this client won't be resent to the other clients, but instead will be queued for the old client. So if you use multiclient, and don't process the messages of all clients, some messages will appear missing.

**Comment by SomberOfShadow** [comment]
@[zmstone](https://github.com/zmstone) 
hi， could you give me hand ?  I see similar issues : https://github.com/eclipse/paho.mqtt.java/issues/367
I want to know whether this issue has been solved?

And MQTT 5.0 java paho has been used in my project , but still does not work .

I will appreciate it if you can give me help.

BRs//Hengtai
